test/common/ssl/ssl_socket_test.cc

```diff
@@ -54,19 +54,127 @@ namespace Ssl {
 
 namespace {
 
-// TODO replace the long parameter list with an options object
-void testUtil(const std::string& client_ctx_yaml, const std::string& server_ctx_yaml,
-              const std::string& expected_digest, const std::string& expected_uri,
-              const std::string& expected_local_uri, const std::string& expected_serial_number,
-              const std::string& expected_subject, const std::string& expected_local_subject,
-              const std::string& expected_peer_cert, const std::string& expected_stats,
-              bool expect_success, const Network::Address::IpVersion version) {
+/**
+ * A base class to hold the options for testUtil() and testUtilV2().
+ */
+class TestUtilOptionsBase {
+public:
+  const std::string& expectedClientCertUri() const { return expected_client_cert_uri_; }
+  const std::string& expectedServerStats() const { return expected_server_stats_; }
+  bool expectSuccess() const { return expect_success_; }
+  Network::Address::IpVersion version() const { return version_; }
+
+protected:
+  TestUtilOptionsBase(bool expect_success, Network::Address::IpVersion version)
+      : expect_success_(expect_success), version_(version) {}
+
+  void setExpectedClientCertUri(const std::string& expected_client_cert_uri) {
+    expected_client_cert_uri_ = expected_client_cert_uri;
+  }
+
+  void setExpectedServerStats(const std::string& expected_server_stats) {
+    expected_server_stats_ = expected_server_stats;
+  }
+
+private:
+  const bool expect_success_;
+  const Network::Address::IpVersion version_;
+
+  std::string expected_server_stats_;
+  std::string expected_client_cert_uri_;
+};
+
+/**
+ * A class to hold the options for testUtil().
+ */
+class TestUtilOptions : public TestUtilOptionsBase {
+public:
+  TestUtilOptions(const std::string& client_ctx_yaml, const std::string& server_ctx_yaml,
+                  bool expect_success, Network::Address::IpVersion version)
+      : TestUtilOptionsBase(expect_success, version), client_ctx_yaml_(client_ctx_yaml),
+        server_ctx_yaml_(server_ctx_yaml) {
+    if (expect_success) {
+      setExpectedServerStats("ssl.handshake");
+    } else {
+      setExpectedServerStats("ssl.fail_verify_error");
```

Comment:
I know this is merged already, but... why is this the default in case of failure instead of `ssl.connection_error`?

Reply:
Because for `testUtil` this is the server stats in all the cases when `expect_success` is false, and for `testUtilV2` it has the same number of cases as `ssl.connection_error`. If you think the default value is important, I can change it in a new PR.